src/main/java/com/ververica/flink/table/gateway/operation/AbstractJobOperation.java

```diff
@@ -102,14 +102,14 @@ public JobStatus getJobStatus() throws SqlGatewayException {
 
 	@Override
 	public void cancelJob() {
-		if (isJobCanceled) {
-			return;
-		}
```

Comment:
Keep this for faster checking

Reply:
that may cause canceling a job multiple times if there are multiple threads

Reply:
I mean keep this checking both inside and outside the synchronized block.